src/ocl/tensorocl.cpp

```diff
@@ -1446,12 +1446,12 @@ void SetTensor(const Handle& handle,
                                           std::multiplies<std::size_t>());
 
         std::size_t wld = 256 < wgd ? 256 : wgd;
-
-        std::string parms = "-DSUBTENSOR_OP_WITH_SCALAR=SUBTENSOR_OP_WITH_SCALAR_SET" +
-                            GetDataTypeKernelParams(dataType);
+        std::stringstream ss;
```

Comment:
I'm just curious - what was the reason to replace std::string with std::stringstream? Potentially `std::stringstream` can be less efficient, more complicated, and we don't need any advanced formatting features and compatibility with stream interface since we pass `std::string` everywhere. (it's not about this particular file, it's almost everywhere)

Reply:
Perhaps @atamazov  can shed more light on which is efficient, but so far the insistence of reviewers has been to prefer `std::stringstream` when constructing strings since it avoid creating many intermediate copies of strings by the `+` operator. Therefore, whenever I touch older code utilizing the `+` operator to concatenate strings I prefer to replace it with stream which can "stream in" substrings without creating copies.